src/cloud_provider/aws/databases/mongodb.rs

```diff
@@ -259,37 +258,30 @@ impl Create for MongoDB {
 }
 
 impl Pause for MongoDB {
-    fn on_pause(&self, _target: &DeploymentTarget) -> Result<(), EngineError> {
+    fn on_pause(&self, target: &DeploymentTarget) -> Result<(), EngineError> {
         info!("AWS.MongoDB.on_pause() called for {}", self.name());
 
-        // TODO how to pause production? - the goal is to reduce cost, but it is possible to pause a production env?
-        // TODO how to pause development? - the goal is also to reduce cost, we can set the number of instances to 0, which will avoid to delete data :)
-
-        Ok(())
+        scale_down_database(target, self, 0)
     }
 
     fn on_pause_check(&self) -> Result<(), EngineError> {
         Ok(())
     }
 
-    fn on_pause_error(&self, _target: &DeploymentTarget) -> Result<(), EngineError> {
+    fn on_pause_error(&self, target: &DeploymentTarget) -> Result<(), EngineError> {
         warn!("AWS.MongoDB.on_pause_error() called for {}", self.name());
 
-        // TODO what to do if there is a pause error?
-
-        Ok(())
+        self.on_pause(target)
```

Comment:
doesn't it create an infinite loop?

Reply:
Going to remove it.
Think it just going to do it twice. It just that it is not very consistent between action & provider, some time we just redo in the xxx_err the same thing than in the exec